### PR TITLE
Bug fix 3.3/agency fix for last confirmed

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: arangod/Agency/Agent.cpp [hunk 1]
```diff
@@ -835,24 +848,37 @@ bool Agent::challengeLeadership() {
 /// Get last acknowledged responses on leader
 query_t Agent::lastAckedAgo() const {
 
-  std::unordered_map<std::string, SteadyTimePoint> lastAcked;
+  decltype(_lastAcked) lastAcked;
```
Comment (neunhoef) [inline arangod/Agency/Agent.cpp]: As discussed, I think this reduces readability, but I do not really care.

File: arangod/Agency/State.cpp [hunk 2]
```diff
@@ -1293,6 +1293,13 @@ index_t State::lastIndex() const {
   return _log.back().index;
 }
 
+// Index of first log entry
+index_t State::firstIndex() const {
+  MUTEX_LOCKER(mutexLocker, _logLock);
+  TRI_ASSERT(!_log.empty());
+  return _log.front().index;
```
Comment (neunhoef) [inline arangod/Agency/State.cpp]: Should we not use `_cur` here instead, which is the same but readily available? Probably this does not matter.

File: arangod/Agency/Agent.cpp (hunk 1 above)
Comment (neunhoef) [inline arangod/Agency/Agent.cpp]: As discussed, I believe this reduces readability, but I do not care very much.

File: arangod/Agency/State.cpp (hunk 2 above)
Comment (kvahed) [reply arangod/Agency/State.cpp]: i would think that this improves readability. you would not have to know, what _cur is. compiler will have a significant acceleration for this i would expect

**Comment by dsonet** [comment]
Can we try this fix sooner? Since we are running clusters, it's really bad that it's in an unstable state.

**Comment by kvahed** [comment]
Ditching this for new backport from devel

**Pull request closed without merging**